### PR TITLE
Add option to group resource list by ore type

### PR DESCRIPTION
**Pull request by okradonkey**
Adds a per-player settings checkbox to group resource list display by ore type.

As requested in #41

### CODE REVIEW

File: resmon.lua [hunk 1]
```diff
@@ -614,13 +614,30 @@ local function site_comparator(left, right)
...
+    elseif left.added_at ~= right.added_at then
+        return left.added_at < right.added_at
+    else
+        return left.name < right.name
+    end
+end
+
 
-local function ascending_by_ratio(sites)
+local function ascending_by_ratio(sites, player)
```
Comment (narc0tiq) [inline resmon.lua]: I just realized I totally mis-named this -- this is `descending_by_ratio`! Derp!

**Comment by narc0tiq** [comment]
Hi, thanks for this, it's awesome!

I'm going to make a couple of changes before merging -- specifically, this is a _sort_ or _order-by_ rather than a _group_, and I do rather want a `group by` capability, too (see #8), so everything will get a rename -- but it's great to have this addition!

Can you think of any other potentially useful sort orders? I'm also considering "order by exact ore count remaining", and maybe "order by estimated time to depletion". If we make it a dropdown-style picker, would you be interested in putting those other sorts together?

Cheers!

File: resmon.lua (hunk 1 above)
Comment (narc0tiq) [reply resmon.lua]: In fact, this ended up being `sites_in_player_order`, with a subsidiary `sites_in_order` that does the actual sorting part based on which comparator you give it. Now we have an easy way to add other possible sort orders via `sites_in_player_order` -- check out ec47f5f7cc90a787788cfb1a310a7208fd72e2f2

**Comment by narc0tiq** [comment]
So, I made it a drop-down:
image

Now we can implement any other sort order we want!

**Comment by okradonkey** [comment]
Awesome, looks great!  I'd be happy to help implement the other sort orders if you haven't already knocked them out by the time you read this.  Honestly, after playing with this in-game for a few hours, I'm starting to wonder if the sort order might eventually earn its own button in the GUI.  But yeah, let's get all the sort choices and logic worked out first; adding a GUI button should be pretty straightforward if you choose to go that route later on.

**Comment by narc0tiq** [comment]
Please do, I'd love the help. I can't steal much more time from my work today. 😂

**Pull request merged**